src/main/java/redis/clients/jedis/Connection.java

```diff
@@ -88,24 +89,34 @@ protected Connection sendCommand(final ProtocolCommand cmd, final String... args
     return sendCommand(cmd, bargs);
   }
 
-  protected Connection sendCommand(final ProtocolCommand cmd, final byte[]... args) {
-    try {
-      connect();
-      Protocol.sendCommand(outputStream, cmd, args);
-      return this;
-    } catch (JedisConnectionException ex) {
-      // Any other exceptions related to connection?
-      broken = true;
-      throw ex;
-    }
+  protected Connection sendCommand(final ProtocolCommand cmd) {
+    return sendCommand(cmd, EMPTY_ARGS);
   }
 
-  protected Connection sendCommand(final ProtocolCommand cmd) {
+  protected Connection sendCommand(final ProtocolCommand cmd, final byte[]... args) {
     try {
       connect();
-      Protocol.sendCommand(outputStream, cmd, new byte[0][]);
+      Protocol.sendCommand(outputStream, cmd, args);
       return this;
     } catch (JedisConnectionException ex) {
+      /*
+       * When client send request which formed by invalid protocol, Redis send back error message
+       * before close connection. We try to read it to provide reason of failure.
+       */
+      try {
+        // Set soTimeout 100ms to avoid long blocking
+        socket.setSoTimeout(100);
+        String errorMessage = Protocol.readErrorLineIfPossible(inputStream);
```

Comment:
Just to test I've changed this code to something like this and test passes:

```java
  */
      try {
        // Set soTimeout 100ms to avoid long blocking
        socket.setSoTimeout(100);
        Protocol.read(inputStream);
      } catch (JedisDataException jde) {
    	  throw new JedisConnectionException(jde.getMessage());
      } catch (Exception e) {
        /*
         * Catch any IOException or JedisConnectionException occurred from InputStream#read and just
         * ignore. This approach is safe because reading error message is optional and connection
         * will eventually be closed.
         */
      }
```

I believe it's much cleaner and it avoids duplicated code. What do you think?